Remove Slack references. Add Gitter References (#2848)
jenkins-infra/jenkins.io: content/sigs/ux/index.adoc

* Discussing UI frameworks
* Discussing impact of UI changes to existing plugins

=== Gitter Channel

> The UX SIG use Gitter as the primary way to discuss any relevant topics. Initially this used to be a CDF Slack channel but a decision was made to move to Gitter in early February 2020. 

Reviewer: Typically we've preferred sentence per line for easier visualization of changes from one version to the next.
Not required, just for your consideration.
```suggestion
The UX SIG uses Gitter to discuss relevant topics.
Initially this used to be a CDF Slack channel but a decision was made to move to Gitter in early February 2020. 
```

The room in Gitter used for this is jenkinsci/ux-sig. 

> You can join the Gitter room via this link:https://gitter.im/jenkinsci/ux-sig?utm_source=share-link&utm_medium=link&utm_campaign=share-link[link].

Reviewer: Simplify link to the gitter room, no need for the utm_campaign or utm_source attributes.
```suggestion
Join the link:https://gitter.im/jenkinsci/ux-sig[Gitter room] for user experience SIG discussions.
```

=== Contributing

> If you want to contribute we recommend that you join the Jenkins UX SIG. You can find details of when the next meetings are planned in the Jenkins Calendar or via the jenkinsci/ux-sig Gitter room.

Reviewer: ```suggestion
If you want to contribute we recommend that you join the Jenkins UX SIG. You can find details of when the next meetings are planned in the Jenkins Calendar or via the link:https://gitter.im/jenkinsci/ux-sig[jenkinsci/ux-sig Gitter room].
```
